File: 1.two-sum/python/src/two_sum.py
class TwoSum:
  def do(self, nums, target):
    map_list = {}

    for i in range(len(nums)):
      current = nums[i]
      diff = target - current

      if current in map_list:
        return [map_list[current], i]

      map_list[diff] = i

File: 1.two-sum/nodejs/src/two_sum.ts
class TwoSum {
  _do(nums: number[], target: number): number[] {
    const map = {}

    for (let i = 0; i < nums.length; i++) {
      const current = nums[i]
      const diff = target - current

      if (current in map) {
        return [map[current], i]
      }

      map[diff] = i
    }

    return []
  }
}

export { TwoSum }
